src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -70,33 +74,24 @@ impl CloudwatchFuture {
 }
 
 impl Future for CloudwatchFuture {
-    type Item = ();
-    type Error = CloudwatchError;
+    type Output = Result<(), CloudwatchError>;
 
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+        let this = self.get_mut();
         loop {
-            match &mut self.state {
+            match &mut this.state {
                 State::DescribeStream(fut) => {
-                    let response = match fut.poll() {
-                        Ok(Async::Ready(res)) => res,
-                        Ok(Async::NotReady) => return Ok(Async::NotReady),
-                        Err(e) => {
-                            if let RusotoError::Service(
-                                DescribeLogStreamsError::ResourceNotFound(_),
-                            ) = e
-                            {
-                                if self.create_missing_group {
-                                    info!("log group provided does not exist; creating a new one.");
-
-                                    self.state = State::CreateGroup(self.client.create_log_group());
-                                    continue;
-                                } else {
-                                    return Err(CloudwatchError::Describe(e));
-                                }
-                            } else {
-                                return Err(CloudwatchError::Describe(e));
-                            }
+                    let response = match ready!(Pin::new(fut).poll(cx)) {
```

Comment:
if this future is `Unpin` (if its boxed it is) you can use `FutureExt::poll_unpin` which may be a bit more clean.